autoload/lsp/lspserver.vim

```diff
@@ -282,7 +282,11 @@ enddef
 
 # send a response message to the server
 def SendResponse(lspserver: dict<any>, request: dict<any>, result: dict<any>, error: dict<any>)
-  var resp: dict<any> = lspserver.createResponse(request.id)
+  if trim(string(request.id), "'") =~ '[^[:digit:]]\+' || trim(string(request.id), "'") == ''
```

Comment:
Do we need a check a for type(request.id) == v:t_string here before the call to trim()?

Reply:
thx review, i am changing the way you suggested.